presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -227,6 +227,10 @@ protected Node visitShowCatalogs(ShowCatalogs node, Void context)
                     .map(name -> row(new StringLiteral(name)))
                     .collect(toList());
 
+            if (rows.isEmpty()) {
+                return simpleQuery(selectList(new LongLiteral("0")));
```

Comment:
Because we want to return 0 as the result, in the case when there are no catalogs to be shown (because the user doesn't have access to any).